ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -455,6 +456,50 @@ private void updateSubviewClipStatus(View subview) {
     }
   }
 
+  @Override
+  public boolean getChildVisibleRect(View child, Rect r, android.graphics.Point offset) {
+    // This is based on the Android ViewGroup implementation, modified to clip child rects
+    // if overflow is set to ViewProps.HIDDEN. This effectively solves Issue #23870 which
+    // appears to have been introduced by FLAG_CLIP_CHILDREN being forced false
+    // regardless of whether clipping is desired.
+    final RectF rect = new RectF();
+    rect.set(r);
+
+    child.getMatrix().mapRect(rect);
+
+    final int dx = child.getLeft() - getScrollX();
+    int dy = child.getTop() - getScrollY();
+
+    rect.offset(dx, dy);
+
+    if (offset != null) {
+      float[] position = new float[2];
+      position[0] = offset.x;
+      position[1] = offset.y;
+      child.getMatrix().mapPoints(position);
+      offset.x = Math.round(position[0]) + dx;
+      offset.y = Math.round(position[1]) + dy;
+    }
+
+    final int width = getRight() - getLeft();
+    final int height = getBottom() - getTop();
+
+    boolean rectIsVisible = true;
+
+    ViewParent parent = getParent();
+    if (parent == null || ViewProps.HIDDEN.equals(mOverflow)) {
+      rectIsVisible = rect.intersect(0, 0, width, height);
+    }
+
+    r.set((int) Math.floor(rect.left), (int) Math.floor(rect.top),
+      (int) Math.ceil(rect.right), (int) Math.ceil(rect.bottom));
+
+    if (rectIsVisible && parent != null) {
+      rectIsVisible = parent.getChildVisibleRect(this, r, offset);
+    }
+    return rectIsVisible;
+  }
+
```

Comment:
Last but not least -- lot's of code duplication here... how about exporting all of this to an external util of some sort?

Reply:
I agree entirely. Was torn on this as well but decided to go with this way to get feedback first. I suppose I can always pass the value of mOverflow or any other dependencies. 

ReactClippingViewGroupHelper.java seems like a good candidate for such a function. I'll take a look.